### PR TITLE
Save original notes on dependent records

### PR DESCRIPTION
**Pull request by yannham**
Save the notes of #52, such that we can close the long-lived and inactive issue without loosing the thoughts.

### CODE REVIEW

File: notes/sum-as-dependent-records.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
+# Notes on sum encoding with dependent types
+
+**This is an archive of thoughts originally written in issue [#54](https://github.com/tweag/nickel/issues/52)**
+
+
+_**Current status**: Intersection and Union types ended up being much more complex than anticipated, on standby for now. We are looking for a simpler way to encode Sum types, this issue presents a possible solution._
+
+The idea would be to encode sums by using Pi and Sigma types: `A+B` would become `Sigma b: Bool, if b then A else B`, which could be constructed with `Pi b: Bool, (if b then A else B) -> Sigma c: Bool, if c then A else B`.
+
+For this, we'd need to implement two new kind of contracts:
```
Comment (Acaccia) [inline notes/sum-as-dependent-records.md:10]: Should we allow contractions in documents ?

File: notes/sum-as-dependent-records.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+ * Dependent functions, `x: A -> \x. B`, which allow the codomain contract to depend on the input, as a reference we have [1] Racket's functions contracts, [2] is also a good reference. There are two challenges/decisions:
+   * _strictness_, a function that ignores it's input may evaluate it anyways if the target contract does
```
Comment (Acaccia) [inline notes/sum-as-dependent-records.md:12]: its

Comment (Acaccia) [reply notes/sum-as-dependent-records.md:12]: anyways

**Review by Acaccia — APPROVED** [review]
There are several contractions in the text, should we allow them for these kind of notes ?

File: notes/sum-as-dependent-records.md (hunk 2 above)
Comment (yannham) [inline notes/sum-as-dependent-records.md:12]:
```suggestion
   * _strictness_, a function that ignores it input may evaluate it anyways if the target contract does
```

Comment (yannham) [inline notes/sum-as-dependent-records.md]:
```suggestion
   * _strictness_, a function that ignores it's input may evaluate it anyway if the target contract does
```

**Comment by aspiwack** [comment]
> There are several contractions in the text, should we allow them for these kind of notes ?

Contractions are always ok in English text. It's really not something that should should worry about.

**Pull request merged**